grub-core/loader/i386/slaunch_skinit.c

```diff
@@ -25,27 +25,32 @@
 #include <grub/dl.h>
 #include <grub/slaunch.h>
 
+static inline grub_uint32_t *get_bootloader_data_addr (struct grub_slaunch_module *mod)
+{
+  grub_uint16_t *ptr = (grub_uint16_t *)mod->addr;
+  return (grub_uint32_t *)(mod->addr + ptr[1]);
```

Comment:
In `relocator_slaunch.c` you have changed the `grub_uint32_t *` to `grub_addr_t`. Why here do we use `grub_uint32_t *` again?

Reply:
Because in [line 46](https://github.com/3mdeb/grub2/pull/7/files#diff-8c3b563ee1b1d55ad329e699ed73ccbaR46) the table is indexed as a `grub_uint32_t` table. Explicit type conversion must be used somewhere, either here or during call to this function.

Reply:
Actually, this should probably be converted into a struct with a known (and extendable) layout to provide some kind of compatibility in the future.